StonehearthEditor/Recipes/RecipesView.cs

```diff
@@ -0,0 +1,701 @@
+            {
+                Regex matchIng = new Regex(IngredientColumnGroup.kIngr + "\\d");
+                for (int i = 0; i < recipesGridView.Columns.Count; i++)
+                {
+                    DataGridViewColumn column = recipesGridView.Columns[i];
+                    bool isText = column.ValueType == typeof(string) || column.ValueType == typeof(int);
+                    bool match = colFilter.Contains(IngredientColumnGroup.kIngr) ? matchIng.IsMatch(column.Name) : isText;
+                    if (match)
+                    {
+                        sb.Append(GetColFilterString(column.Name, searchTerm));
+                        if (i < recipesGridView.Columns.Count - 1)
+                        {
+                            sb.Append(" OR "); // to aggregate multiple columns into the search filter query
+                        }
+                    }
+                }
+            }
+            else
+            {
+                // Filter by specific columns
+                sb.Append(GetColFilterString(colFilter, searchTerm));
+            }
+
+            mDataTable.DefaultView.RowFilter = sb.ToString();
+        }
+
+        private void recipesGridView_KeyDown(object sender, KeyEventArgs e)
+        {
+            // Delete cell on del
+            if (e.KeyCode == Keys.Delete)
+            {
+                DeleteCurrentCell();
+            }
+
+            // Save on ctrl+s
+            else if (e.Control && e.KeyCode == Keys.S)
+            {
+                SaveModifiedFiles();
+            }
+
+            // Paste multiple cells on ctrl+v
+            else if (e.Control && e.KeyCode == Keys.V)
+            {
+                if (recipesGridView.SelectedCells.Count == 0)
+                {
+                    return;
+                }
+
+                Tuple<DataGridViewCell, DataGridViewCell> boundaryCells = GetSelectedCellsBoundaries();
+                DataGridViewCell startCell = boundaryCells.Item1;
+                DataGridViewCell endCell = boundaryCells.Item2;
+                int startRow = startCell.RowIndex;
+                int startCol = startCell.ColumnIndex;
+
+                int rowIndex = startRow;
+                string s = Clipboard.GetText();
+                string[] lines = s.Split(new string[] { System.Environment.NewLine }, StringSplitOptions.None);
+                int lineIndex = 0;
+                while (lineIndex < lines.Count())
+                {
+                    string line = lines[lineIndex];
+                    if (rowIndex < recipesGridView.Rows.Count)
+                    {
+                        string[] cells = line.Split('\t');
+                        int colIndex = startCol;
+                        for (int i = 0; i < cells.Length && colIndex < recipesGridView.Columns.Count; i++)
+                        {
+                            SetGridValue(colIndex, rowIndex, cells[i]);
```

Comment:
If this fails for any cell, we still update the other cells. As a user, I would expect this to be atomic, and ideally undoable.